### PR TITLE
Remove index append executor

### PR DESCRIPTION
**Pull request by Bhupender-Y**
**Change log description**  
Remove index append executor which we added for index append and use existing low priority executor for append on index segment.

**Purpose of the change**  
N/A

**What the code does**  
Remove index append executor which we added for index append and use existing low priority executor for append on index segment.

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java [hunk 1]
```diff
@@ -38,29 +38,28 @@
...
     public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor, StreamSegmentStore store) {
-        this.indexSegmentUpdateExecutor = indexSegmentUpdateExecutor;
         this.store = store;
+        appendProcessor = new MultiKeyLatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
     }
 
     /**
      * Appends index segment on a separate thread.
      * @param segmentName  segment name.
+     * @param maxAllowedSize  Maximum allowed event size.
```
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java:52]: Not max, Only allowed size.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java [hunk 2]
```diff
@@ -38,29 +38,28 @@
...
      */
-    protected void processAppend(String segmentName) {
+    protected void processAppend(String segmentName, long maxAllowedSize) {
         // Not updating index segment for transient and transaction type.
         if (isTransientSegment(segmentName) || isTransactionSegment(segmentName)) {
             return;
         }
-        LatestItemSequentialProcessor<String> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
-        latestItemProcessor.updateItem(segmentName);
+        appendProcessor.updateItem(segmentName, maxAllowedSize);
```
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java:59]: AllowedSize is not being used/checked. If it's wrong then we don't need to add an item.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java [hunk 3]
```diff
@@ -38,29 +38,28 @@
...
     }
 
-    private void handleIndexAppend(String segmentName) {
-        //TODO : Update the index segment Attribute as needed going forward. If eventcount for main segment is null then we are passing it as 0 in index segment.
+    private void handleIndexAppend(String segmentName, long maxAllowedSize) {
```
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java:62]: This logic is incorrect. 
The append needs to be conditional on the count being higher. This means the `attributes` below needs to be setting the value from the `info.getAttributes()`.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java (hunk 1 above)
Comment (Bhupender-Y) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java]: Done.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java (hunk 3 above)
Comment (Bhupender-Y) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java:62]: This logic is handled in PR #7199 .

**Review by tkaitchuck — APPROVED** [review]

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**